sdk/all/src/test/java/io/opentelemetry/sdk/ScopeConfiguratorTest.java

```diff
@@ -0,0 +1,217 @@
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.sdk.common.InstrumentationScopeInfo;
+import io.opentelemetry.sdk.logs.LoggerConfig;
+import io.opentelemetry.sdk.logs.SdkLoggerProvider;
+import io.opentelemetry.sdk.logs.data.LogRecordData;
+import io.opentelemetry.sdk.logs.export.SimpleLogRecordProcessor;
+import io.opentelemetry.sdk.metrics.MeterConfig;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.testing.exporter.InMemoryLogRecordExporter;
+import io.opentelemetry.sdk.testing.exporter.InMemoryMetricReader;
+import io.opentelemetry.sdk.testing.exporter.InMemorySpanExporter;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.TracerConfig;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.Test;
+
+class ScopeConfiguratorTest {
+
+  private final InMemoryLogRecordExporter logRecordExporter = InMemoryLogRecordExporter.create();
+  private final InMemoryMetricReader metricReader = InMemoryMetricReader.create();
+  private final InMemorySpanExporter spanExporter = InMemorySpanExporter.create();
+
+  /** Disable "scopeB". All other scopes are enabled by default. */
+  @Test
+  void disableScopeB() {
+    OpenTelemetrySdk sdk =
+        OpenTelemetrySdk.builder()
+            .setTracerProvider(
+                SdkTracerProvider.builder()
+                    .addSpanProcessor(SimpleSpanProcessor.create(spanExporter))
+                    .addTracerConfiguratorCondition(nameEquals("scopeB"), TracerConfig.disabled())
+                    .build())
+            .setMeterProvider(
+                SdkMeterProvider.builder()
+                    .registerMetricReader(metricReader)
+                    .addMeterConfiguratorCondition(nameEquals("scopeB"), MeterConfig.disabled())
+                    .build())
+            .setLoggerProvider(
+                SdkLoggerProvider.builder()
+                    .addLogRecordProcessor(SimpleLogRecordProcessor.create(logRecordExporter))
+                    .addLoggerConfiguratorCondition(nameEquals("scopeB"), LoggerConfig.disabled())
+                    .build())
+            .build();
+
+    simulateInstrumentation(sdk);
+
+    // Collect all the telemetry. Ensure we don't see any from scopeB, and that the telemetry from
+    // scopeA and scopeC is valid.
+    assertThat(spanExporter.getFinishedSpanItems())
+        .satisfies(
+            spans -> {
+              Map<InstrumentationScopeInfo, List<SpanData>> spansByScope =
+                  spans.stream()
+                      .collect(Collectors.groupingBy(SpanData::getInstrumentationScopeInfo));
+              assertThat(spansByScope.get(InstrumentationScopeInfo.create("scopeA"))).hasSize(1);
```

Comment:
What would you say about making 3 constant `InstrumentationScopeInfo` instances for these (`SCOPE_A`, `SCOPE_B`, and `SCOPE_C`)? I think it might help readability a little bit, since these are used throughout this class. 👍🏻

Reply:
Yeah I should do that.